scripts/generate-json-schema.ts

```
@@ -205,11 +215,40 @@ async function generateJsonSchema() {
 
+	// Check if existing root schema is identical to avoid unnecessary writes
+	const existingRootSchema = await Result.pipe(
+		readFile(SCHEMA_FILENAME),
+		Result.map(content => JSON.parse(content) as unknown),
+		Result.unwrap(''),
+	);
+
+	const isSchemaChanged = !Bun.deepEquals(existingRootSchema, schemaObject, true);
+
+	if (!isSchemaChanged) {
+		logger.info('✓ Root schema is up to date, skipping generation');
```

ESLint

[error] Unsafe call of a(n) `error` type typed value.

(ts/no-unsafe-call)